awscli/customizations/cloudformation/artifact_exporter.py

```diff
@@ -94,6 +94,51 @@ def parse_s3_url(url,
                      "{0}".format(url))
 
 
+def get_nested_property_value(resource_dict, property_name):
```

Comment:
For both the `get` and `set` methods, would it be possible to use [JMESPath](https://github.com/jmespath/jmespath.py) instead? It is a JSON query language we use in the CLI codebase. Specifically, we could replace calls to `get_nested_property_value` to: `jmespath.search(property_name, resource_dict)` and calls to `set_nested_property_value` to `botocore.utils.set_value_from_jmespath`. If we use these functions instead we should be able to cut down the amount of code we need to add and not have to worry about explicitly testing the helper functions as the new functions we would be calling are already tested.

Reply:
Sure, let me look into what that will take.  I've used jmespath with the --query output from the CLI, but never worked with the jmespath library.

Reply:
@kyleknap, I've refactored this to use the recommended methods.  Thanks for pointing that out, that significantly simplified this implementation!